server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -395,6 +398,24 @@ private void tryAllHandlers(final RestRequest request, final RestChannel channel
                         return;
                     }
                 } else {
+
+                    final Optional<String> authHeader = request.getHeaders()
+                        .getOrDefault("Authorization", Collections.emptyList())
+                        .stream()
+                        .findFirst();
+
+                    try {
+                        MyShiroModule.authenticateViaAuthorizationHeader(authHeader);
+                    } catch (final Exception e) {
```

Comment:
Exception is general here. The method throws an `AuthenticationException`, should we be catching the more specific exception here?

Reply:
Yes! This code is bad - its demonstrative for design/prototype review but not meant to be 'actual' implementation.

It will get scrubbed with unit tests and proper error handling